src/query/sql/src/planner/semantic/name_resolution.rs

```diff
@@ -56,7 +56,7 @@ pub fn normalize_identifier<'a>(
         ident.clone()
     } else {
         Identifier {
-            name: ident.name.to_lowercase(),
+            name: ident.name.clone(),
```

Comment:
I'm afraid this issue is more deep than this.

```sql
CREATE VIEW view_1 AS WITH source as (SELECT 1) SELECT * from source;

SELECT * from view_1;

CREATE VIEW view_2 AS WITH source as (SELECT 1) SELECT * from source;

SELECT * from view_2;

select * from VIEW_1 join view_2;
```

Reply:
Filed an issue: https://github.com/datafuselabs/databend/issues/9049

Better to fix them in different PRs.